client/src/components/relatedItems/RelatedItems.jsx

```diff
@@ -1,19 +1,120 @@
-import React from 'react';
+import React, { Component } from "react";
+import Item from './Item.jsx';
+// import './relatedItems.css';
 
 class RelatedItems extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
+      current_card: 1
+    }
+  }
+
+  componentDidMount() {
+    let first_card_clone = this.card_container.children[0].cloneNode(true);
+    let last_card_clone = this.card_container.children[this.card_container.children.length - 1].cloneNode(true);
+
+    this.card_container.insertBefore(last_card_clone, this.card_container.children[0]);
+    this.card_container.append(first_card_clone);
+    this.card_container.style.transitionDuration = '0s';
+    this.card_container.style.transform = `translate(-${350}px)`;
+  }
+
+  handle_next = () => {
+    console.log('Click triggered');
+    if (this.state.current_card < this.card_container.children.length - 1) {
+      let new_current_card = this.state.current_card + 1;
+
+      this.setState({ current_card: new_current_card }, () => {
+        this.card_container.style.transitionDuration = '0.5s';
+        this.card_container.style.transform = `translate(-${350 * this.state.current_card}px)`;
+
+        if (this.state.current_card === this.card_container.children.length - 1) {
+          setTimeout(() => {
+            this.card_container.style.transitionDuration = '0s';
+            this.card_container.style.transform = `translate(-${350}px)`;
+
+            this.setState({current_card: 1});
+          }, 500);
+
+        }
+
+      });
+
+    } else {
+      return;
+    }
+
+
+  }
+
+  handle_previous = () => {
+    if (this.state.current_card > 0) {
+      let new_current_card = this.state.current_card - 1;
+
+      this.setState({ current_card: new_current_card }, () => {
+        this.card_container.style.transitionDuration = '0.5s';
+        this.card_container.style.transform = `translate(-${350 * this.state.current_card}px)`;
+
+        if (this.state.current_card === 0) {
+          setTimeout(() => {
+            this.card_container.style.transitionDuration = '0s';
+            this.card_container.style.transform = `translate(-${350 * (this.card_container.children - 2)}px)`;
+
+            this.setState({current_card: this.card_container.children.length - 2});
+          }, 500);
+
+        }
+
+      });
+
+    } else {
+      return;
+    }
+
   }
 
   render() {
     return (
       <div>
-        <h2>Related Products</h2>
-        <div>CardsList</div>
-        <div>YourOutfitList</div>
+        <h1>Related Products</h1>
+        <div>
+        <button onClick={this.handle_previous}>Previous</button>
+        <button onClick={this.handle_next}>Next</button>
```

Comment:
I will replace the buttons eventually. I just have it for now to move the cards in carousel left and right

Reply:
Got it. Can you also fix the failing tests?